lib/ansible/plugins/filter/core.py

```diff
@@ -277,7 +277,12 @@ def get_encrypted_password(password, hashtype='sha512', salt=None, salt_size=Non
 
 
 def to_uuid(string):
-    return str(uuid.uuid5(UUID_NAMESPACE_ANSIBLE, str(string)))
+    return to_uuid5(string, namespace=UUID_NAMESPACE_ANSIBLE)
```

Comment:
this changes the return value and is not backwards compatible

Reply:
Unless there is some mistake two lines below, the return value should be identical. However, the unit tests did not detect the issue you mention.

Would you help me reproduce please?

Reply:
I think this will be ok since `to_uuid()` is returning a string. But this does look wrong on first glance.

Reply:
i was reffering to the `str()` .. but i missed it was added in the new function .. though it should be `to_text()` in any case

Reply:
Excuse the ignorance. On line 285, would it be

```Python
    return to_text(uuid.uuid5(uuid_namespace, str(string)))
```

or
```Python
    return to_text(uuid.uuid5(uuid_namespace, to_text(string)))
```
?

(in the second case, all occurrencies of `str()` have been replaced with `to_text()`)